PolygonEnemy/src/test/java/com/github/sef24sp4/polygonenemy/PolygonEnemyTest.java

```diff
@@ -0,0 +1,52 @@
+package com.github.sef24sp4.polygonenemy;
+
+import com.github.sef24sp4.common.metadata.IGameMetadata;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class PolygonEnemyTest {
+
+	private PolygonEnemy polygonEnemy;
+
+	@BeforeEach
+	void setUp() {
+		this.polygonEnemy = new PolygonEnemy(5);
+	}
+
+	@AfterEach
+	void tearDown() {
+	}
+
+	@Test
+	void getEdges() {
+		assertEquals(5, this.polygonEnemy.getEdges());
+	}
+
+	@Test
+	void getMaxHealth() {
+		assertEquals(5, this.polygonEnemy.getMaxHealth());
+	}
+
+	@Test
+	void getHealth() {
+		assertEquals(5, this.polygonEnemy.getHealth());
+	}
+
+	@Test
+	void getAttackDamage() {
+		assertEquals(5, this.polygonEnemy.getAttackDamage());
+	}
+
+	@Test
+	void collide() {
+
+	}
+
+	@Test
+	void getMetadata() {
+		assertInstanceOf(IGameMetadata.class, this.polygonEnemy.getMetadata());
```

Comment:
Maybe also make an assertion that the metadata contains the correct information.